lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -35,6 +40,46 @@ def __init__(self, apis, concrete_artifacts_manager):
         self._apis = apis
         self._concrete_art_mgr = concrete_artifacts_manager
 
+    def _get_collection_versions_with_api_order(self, requirement, api_lookup_order):
+        # type: (Candidate, Iterable[GalaxyAPI]) -> Iterable[Tuple[GalaxyAPI, str]]
+        """Helper for get_collection_versions. Yield api, version pairs for all APIs and reraise the last error if no valid API was found."""
+        found_api = last_error = None
+
+        for api in api_lookup_order:
```

Comment:
Looking closer, I wonder why this doesn't use `self._apis`. If it's different, then this method definitely doesn't belong in this class.

Reply:
Fixed.